mbuild/formats/gomc_conf_writer.py

```diff
@@ -0,0 +1,4065 @@
+        self.MEMC_1Freq = default_input_variables_dict['MEMC-1Freq'][self.ensemble_type]
+        self.IntraMEMC_2Freq = default_input_variables_dict['IntraMEMC-2Freq'][self.ensemble_type]
+        self.MEMC_2Freq = default_input_variables_dict['MEMC-2Freq'][self.ensemble_type]
+        self.IntraMEMC_3Freq = default_input_variables_dict['IntraMEMC-3Freq'][self.ensemble_type]
+        self.MEMC_3Freq = default_input_variables_dict['MEMC-3Freq'][self.ensemble_type]
+
+        self.ExchangeVolumeDim = default_input_variables_dict['ExchangeVolumeDim']
+        self.MEMC_DataInput = default_input_variables_dict['MEMC_DataInput']
+
+        # auto calculate the best EqSteps (number of Equilbrium Steps) and Adj_Steps (number of AdjSteps Steps)
+        set_max_steps_equib = self.EqSteps #1 * 10 ** 6
+
+        if RunSteps / 10 >= set_max_steps_equib and RunSteps / 10 >= 1:
+            self.EqSteps = int(set_max_steps_equib)
+        elif RunSteps / 10  >= 1:
+            self.EqSteps = int(RunSteps / 10)
+        else:
+            self.EqSteps = int( 1)
+
+        set_max_steps_adj = self.AdjSteps #1000
+        if RunSteps / 10 >= set_max_steps_adj and RunSteps / 10 >= 1:
+            self.AdjSteps = int(set_max_steps_adj)
+        elif RunSteps / 10 >= 1:
+            self.AdjSteps = int(RunSteps / 10)
+        else:
+            self.AdjSteps = int(1)
+
+
+
+        # auto calculate the best RestartFreq  for the number of RunSteps
+        set_max_steps_RestartFreq  = self.RestartFreq[1] #1 * 10 ** 6
+
+        if RunSteps / 10 >= set_max_steps_RestartFreq and RunSteps / 10 >= 1:
+            self.RestartFreq[1] = int(set_max_steps_RestartFreq)
+        elif RunSteps / 10  >= 1:
+            self.RestartFreq[1] = int(RunSteps / 10)
+        else:
+            self.RestartFreq[1] = int( 1)
+
+        # auto calculate the best CheckpointFreq  for the number of RunSteps
+        set_max_steps_CheckpointFreq  = self.CheckpointFreq[1] # 1 * 10 ** 6
+
+        if RunSteps / 10 >= set_max_steps_CheckpointFreq and RunSteps / 10 >= 1:
+            self.CheckpointFreq[1] = int(set_max_steps_CheckpointFreq)
+        elif RunSteps / 10  >= 1:
+            self.CheckpointFreq[1] = int(RunSteps / 10)
+        else:
+            self.CheckpointFreq[1] = int( 1)
+
+        # auto calculate the best CoordinatesFreq  for the number of RunSteps
+        set_max_steps_CoordinatesFreq  = self.CoordinatesFreq[1] # 1 * 10 ** 6
+
+        if RunSteps / 10 >= set_max_steps_CoordinatesFreq and RunSteps / 10 >= 1:
+            self.CoordinatesFreq[1] = int(set_max_steps_CoordinatesFreq)
+        elif RunSteps / 10  >= 1:
+            self.CoordinatesFreq[1] = int(RunSteps / 10)
+        else:
+            self.CoordinatesFreq[1] = int( 1)
+
+        # auto calculate the best ConsoleFreq  for the number of RunSteps
+        set_max_steps_ConsoleFreq  = self.ConsoleFreq[1] #1 * 10 ** 4
+
+        if RunSteps / 10 >= set_max_steps_ConsoleFreq and RunSteps / 10 >= 1:
```

Comment:
All of this math from 1345-1445 has the same basic algorithm, this should be made into a function that is then called to return the proper `ConsoleFreq`, etc

Seems like they all have the same conditionals, just a different max_steps parameter.

Reply:
Fixed what I could, compressing them all to 3 functions